lib/semantics/expression.cc

```diff
@@ -136,23 +141,46 @@ common::IfNoLvalue<MaybeExpr, WRAPPED> TypedWrapper(
 class ArgumentAnalyzer {
 public:
   explicit ArgumentAnalyzer(ExpressionAnalyzer &context) : context_{context} {}
-  bool success() const { return success_; }
+  bool fatalErrors() const { return fatalErrors_; }
   ActualArguments &&GetActuals() {
-    CHECK(success_);
+    CHECK(!fatalErrors_);
     return std::move(actuals_);
   }
-  template<typename T> void Analyze(const T &x) {
-    actuals_.emplace_back(context_.Analyze(x));
-    success_ &= actuals_.back().has_value();
+  Expr<SomeType> GetAsExpr(std::size_t i) const {
```

Comment:
Could this return a `const &` and avoid the copy, or might the results outlive the source object?

Reply:
These get passed as r-value references to NumericOperation, etc. so we would need a copy anyway.